 <launch>
  <!-- Arguments -->
  <arg name="scan_topic"     default="scan"/>
  <!-- <arg name="initial_pose_x" default="8.314"/>
  <arg name="initial_pose_y" default="-2.489"/>
  <arg name="initial_pose_a" default="2.517"/> -->
  <arg name="use_map_topic" default="true"/>

  <arg name="initial_pose_x" default="0.500"/>
  <arg name="initial_pose_y" default="-8.501"/>
  <arg name="initial_pose_a" default="3.125"/>

  <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find ohmni_nav)/maps/hopee_edit_again.yaml"/> -->
  <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find ohmni_nav)/maps/chartlab_5cm.yaml"/> -->

  <node name="map_server" pkg="map_server" type="map_server" args="$(find ohmni_nav)/maps/hope_level2.yaml"/>

  <!-- <node name="tf_baseFootprint_lidar" pkg="tf" type="static_transform_publisher" args="0.06 0.0 0.45 0 0 3.14 base_link laser 1000"/> -->
 
  <!-- AMCL -->
 <node pkg="amcl" type="amcl" name="amcl" output="screen">
    <param name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <param name="initial_pose_a" value="$(arg initial_pose_a)"/>

    <param name="use_map_topic" value="$(arg use_map_topic)"/>
    <!-- <param name="first_map_only" value="$(arg use_map_topic)"/> -->
    <!-- Publish scans from best pose at a max of 10 Hz -->
    <param name="odom_model_type" value="diff"/>
    <param name="odom_alpha5" value="0.1"/>

    <param name="gui_publish_rate" value="10.0"/>
    <param name="laser_max_beams" value="60"/>
    <param name="laser_max_range" value="12.0"/>
    <param name="min_particles" value="500"/>
    <param name="max_particles" value="2000"/>

    <!-- odometry model -->
    <param name="kld_err" value="0.05"/>
    <param name="kld_z" value="0.99"/>
    <param name="odom_alpha1" value="2.5"/>
    <param name="odom_alpha2" value="2.0"/>
    <!-- translation std dev, m -->
    <param name="odom_alpha3" value="1.0"/>
    <param name="odom_alpha4" value="1.0"/>

    <!-- laser scanner model -->
    <param name="laser_z_hit" value="0.9"/>
    <param name="laser_z_short" value="0.05"/>
    <param name="laser_z_max" value="0.05"/>
    <param name="laser_z_rand" value="0.5"/>
    <param name="laser_sigma_hit" value="0.2"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_model_type" value="likelihood_field"/>
    <param name="laser_likelihood_max_dist" value="8.0"/>

    <param name="update_min_d" value="0.25"/>
    <param name="update_min_a" value="0.1"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="resample_interval" value="1"/>
    <!-- Increase tolerance because the computer can get quite busy -->
    <param name="transform_tolerance" value="1.0"/>
    <param name="recovery_alpha_slow" value="0.0"/>
    <param name="recovery_alpha_fast" value="0.0"/>
    <remap from="scan" to="$(arg scan_topic)"/>
  </node>

</launch>